slsDetectorServers/slsDetectorServer/src/common.c

```diff
@@ -370,12 +402,177 @@ int verifyChecksum(char *mess, char *clientChecksum, MD5_CTX *c, char *msg) {
+        return FAIL;
+    }
+    if (executeCommand(cmd, retvals, logDEBUG1) == FAIL) {
+        snprintf(mess, MAX_STR_LENGTH,
+                 "Could not copy detector server (respawning). %s\n", retvals);
+        LOG(logERROR, (mess));
+        return FAIL;
+    }
+    LOG(logINFO, ("\tinittab: updated for respawning\n"));
+
+#endif
+
+    // sync
+    strcpy(cmd, "sync");
+    if (executeCommand(cmd, retvals, logDEBUG1) == FAIL) {
+        snprintf(mess, MAX_STR_LENGTH,
+                 "Could not copy detector server (sync). %s\n", retvals);
+        LOG(logERROR, (mess));
+        return FAIL;
+    }
+    LOG(logINFO, ("\tsync\n"));
+    return OK;
+}
+
+int writeBinaryFile(char *mess, char *fname, char *buffer,
+                    const uint64_t filesize, char *errorPrefix) {
+    LOG(logINFO, ("\tWriting Detector Server Binary...\n"));
+
+    FILE *fp = fopen(fname, "wb");
+    if (fp == NULL) {
+        sprintf(mess,
+                "Could not %s. (opening file to write(%s). "
+                "Maybe it is being used? Try another name?\n",
+                errorPrefix, fname);
+        LOG(logERROR, (mess));
+        return FAIL;
+    }
+
+    size_t bytesWritten = 0;
+    size_t unitSize = 128;
+    int oldProgress = 0;
+
+    while (bytesWritten < filesize) {
+        // print progress
+        int progress = (int)(((double)(bytesWritten) / filesize) * 100);
+        if (oldProgress != progress) {
+            printf("%d%%\r", progress);
+            fflush(stdout);
+            oldProgress = progress;
+        }
+
+        // for less than 128 bytes
+        ssize_t writeSize = unitSize;
+        if ((unitSize + bytesWritten) > filesize) {
+            writeSize = filesize - bytesWritten;
+        }
+        size_t bytes = fwrite((char *)buffer + bytesWritten, 1, writeSize, fp);
+
+        // write
+        if (bytes != (size_t)writeSize) {
+            sprintf(mess,
+                    "Could not %s. Expected to write %lu "
+                    "bytes, wrote %lu bytes). No space left? \n",
+                    errorPrefix, (long unsigned int)filesize,
+                    (long unsigned int)bytesWritten);
+            LOG(logERROR, (mess));
+            return FAIL;
+        }
+        bytesWritten += bytes;
+        LOG(logDEBUG1,
+            ("bytesWritten:%lu filesize:%lu\n", bytesWritten, filesize));
+    }
+    if (fclose(fp) != 0) {
+        sprintf(mess, "Could not %s. (closing file pointer)\n", errorPrefix);
+        LOG(logERROR, (mess));
+        return FAIL;
+    }
+    LOG(logINFO, ("\tWritten binary to %s (%lu bytes)\n", fname,
+                  (long unsigned int)bytesWritten));
+    return OK;
+}
+
+int moveBinaryFile(char *mess, char *dest, char *src, char *errorPrefix) {
+    char cmd[MAX_STR_LENGTH] = {0};
+    char retvals[MAX_STR_LENGTH] = {0};
+
+    // one can move into the current process binary (will not interfere in
+    // kernel mode)
+
+    char *format = "mv %s %s";
```

Comment:
This could be put directly in the snprintf since it's not reused. Advantage is that the warning can be turned on

Reply:
will check it out

Reply:
ok done